### ISSUE TITLE
Open In Internet Explorer Extension for Chrome - terminal services

### ISSUE THREAD

**Issue by AssutaSystem** [andy-portmen/native-client#9]
Hello
I would like to install the extension on a terminal server which is publishing Google Chrome.
Is it possible ton install the extension for all users?
I can install the extension with admin right, can it run with standard users permissions (non-admin)?

Thanks for your help
Yoram

**Comment by andy-portmen** [comment]
Yes, this seems like doable. Download the whole project from https://github.com/andy-portmen/native-client/archive/master.zip

Unzip it in a local directory. Since you only have interest in Windows version, go to the windows folder and open the `install.bat` file with a text editor.

1. Change all the `HKCU` with `HKLM`. This will install the registries for all users instead of the current user.
2. Change `%LocalAPPData%` with an absolute path of a directory that is accessible by all users (this is where the actual NodeJS application is being installed) (make sure this directory is not writeable by other users just in case!)

Notes:
1. If you already have installed the native client you need to delete all the `HKCU`s first. Simply run `uninstall.bat` before modifying the `install.bat`
2. You need to run the script with **administration** permission since `HKLM` is used.

**Comment by andy-portmen** [comment]
> I can install the extension with admin right, can it run with standard users permissions (non-admin)?

So if the native client is installed with `HKLM`, it can be accessed by all users on the same machine even if they have no admin permission. The extension itself also does not require admin permission like other normal extensions. So this should work.

**Comment by AssutaSystem** [comment]
Hello,

I followed your instructions but I was not able to install the software.
I modified the install.bat file as you explained me and I received the following error message:

[cid:image003.png@01D2CA32.D1ED4890]

Here is my modified install.bat:

@echo off

echo .. Writting to Chrome Registry
echo .. Key: HKLM\Software\Google\Chrome\NativeMessagingHosts\com.add0n.node
REG ADD "HKLM\Software\Google\Chrome\NativeMessagingHosts\com.add0n.node" /ve /t REG_SZ /d "%LocalAPPData%\com.add0n.node\manifest-chrome.json" /f

echo .. Writting to Firefox Registry
echo .. Key: HKLM\SOFTWARE\Mozilla\NativeMessagingHosts\com.add0n.node
for %%f in ("%LocalAPPData%") do SET SHORT_PATH=%%~sf
REG ADD "HKLM\SOFTWARE\Mozilla\NativeMessagingHosts\com.add0n.node" /ve /t REG_SZ /d "%SHORT_PATH%\com.add0n.node\manifest-firefox.json" /f

pushd "%~dp0"
cd app
..\node.exe install.js "C:\Program Files (x86)\Google\Chrome\IEButton"

pause

Thanks for your help

Best Regards,

Yoram Knobloch
Technological Projects Team
IT Department
Tel: +972 3 7644137
Cell: +972 54 335 8944
[LogoEng]

**Comment by andy-portmen** [comment]
I still see `%LocalAPPData%` in the batch file. Please use GitHub to post images. Images from email are not being displayed in GitHub

**Comment by AssutaSystem** [comment]
I have corrected the mistake.
But still receive an error message. 
Please see the screenshot.
![image](https://cloud.githubusercontent.com/assets/28600411/25937007/b6af6f38-3632-11e7-84b8-00af0c85795e.png)
Thanks for your help
Yoram

**Comment by andy-portmen** [comment]
I think that's a symlink issue. Can you use https://github.com/andy-portmen/native-client/releases/download/0.3.8/windows.zip instead of https://github.com/andy-portmen/native-client/archive/master.zip

Just apply the same modifications and let me know if you still get this error.

**Comment by AssutaSystem** [comment]
Hello
Thanks for all your help.
I managed to in install the Add-On on my servers, it works fine.
I used GPO to configure the Add-On for all users and installed the EXE file on all the servers.
However, I still have a little issue:
The first time a user open Google Chrome and the Add-On is installed by GPO, the Add-On web site shows up.
Is it possible to prevent this?

Thanks in advance
Yoram
![erroraddon](https://cloud.githubusercontent.com/assets/28600411/26538774/72a93f3a-4451-11e7-9643-45f6064f3e2c.jpg)

**Comment by andy-portmen** [comment]
Can GPO install unpacked extensions?

**Comment by andy-portmen** [comment]
There are basically two ways to prevent the FAQs page to pop up after installation.

1. Install a version (unpacked version) that does not contain the code to open the FAQs page
2. Set the "version" preference in the add-on storage. So that extension bypasses FAQs page opening.

**Comment by AssutaSystem** [comment]
Hello

Thanks for your answer
The second option could be interesting...
How should I proceed?

Yoram

**Comment by andy-portmen** [comment]
It requires running something like
```js
chrome.storage.local.set({version: '0.1.0'});
```
in the add-on context. 

So basically the add-on stores its version once the extension is initialized and if there is no stored version, the FAQs page is opened once to direct new users to this page for instruction.

I am not really familiar with GPO so I am not sure if this is doable or not. If not the first option seems like doable to me.

**Comment by AssutaSystem** [comment]
Hello
I made some research regarding deploying unpacked extensions by group policy.
It seems to be possible.
I need to use an ExtensionInstallForcelist policy.
That is what I already use. It is a list of ID and WebStore URL paired.
If we decide to publish an unpacked extension, aka without an Official WebStore URL, we need to provide a CRX or XML file which contains the same data that the URL provides.
Can you please send me the exentsion sources and help me with the CRX file
Thanks in advance
Yoram

**Comment by andy-portmen** [comment]
Use http://chrome-extension-downloader.com/ to download the CRX. CRX is basically a zip archive. So use a zip tool to extract the content. After modifying the files, use chrome://extensions/ to create a new CRX (turn developer mode on to see the option).

To remove the FAQs page opening remove the block under `// FAQs & Feedback` comment.

**Comment by AssutaSystem** [comment]
Hello
I tried to manage with a custom extension and install it without calling Google Play Store.
I have downloaded the CRX file and  modified the code, I re-pack the extension and published the new CRX file on my IIS server.
But Google Chrome is not allowing me to load an unpacked extension, event after I enabled "dev mode"
How can you help me?

Thanks
Yoram

**Comment by andy-portmen** [comment]
> But Google Chrome is not allowing me to load an unpacked extension, event after I enabled "dev mode"

Do you get an error message?

**Comment by AssutaSystem** [comment]
I did not got any error message...
The add-on is not installed...
Maybe I could approach this issue differently:
If the advertisement shows up only once, I suppose that you are checking a file or a registry key.
I can put this file or registry key by GPP and this could solve the issue.
What do you think? Is it possible?

Thanks
Yoram

**Comment by andy-portmen** [comment]
Have you seen:

http://dev.chromium.org/administrators/policy-list-3#ExtensionInstallForcelist

> For Windows instances that are not joined to an Active Directory domain, forced installation is limited to apps and extensions listed in the Chrome Web Store.

The easiest way to handle this is to create a Chrome Webstore account and upload your CRX there. Make sure to mark your listing as [private](https://support.google.com/chrome/a/answer/2663860?hl=en). However, there are some other solutions recommended on https://stackoverflow.com/questions/30233529/chrome-local-extension-via-group-policy

Also take a look at
https://www.tenfold.com/support-center/deploying-chrome-extension-using-group-policy/

>  can put this file or registry key by GPP and this could solve the issue.

I am not really familiar with GPO but to prevent the  FAQs page from being displayed you need to write to the extension's storage as explained in https://github.com/andy-portmen/native-client/issues/9#issuecomment-304655595

**Comment by andy-portmen** [comment]
This seems like a better solution:

> Force-installing via master_preferences

https://support.google.com/chrome/a/answer/188453?hl=en
